### PR TITLE
Fix mem2reg of lexical enum stack locations

### PR DESCRIPTION
**Pull request by meg-gupta**
When we have a non trivial enum lexical stack location, and we store a none value trivially, after mem2reg we can end up with an ownership error because of the leaking move_value [lexical].

This change avoids creating move_value [lexical] for trivial stores.

Do the same when we have store_borrow of trivial values, even though there
is no ownership error in this case, handle it like store for consistency

### CODE REVIEW

File: lib/SILOptimizer/Transforms/SILMem2Reg.cpp [hunk 1]
```diff
@@ -1164,7 +1178,8 @@ SILInstruction *StackAllocationPromoter::promoteAllocationInBlock(
                      /*isStorageValid=*/true};
       // The current store is now the lastStoreInst.
       lastStoreInst = sbi;
-      if (lexicalLifetimeEnsured(asi)) {
+      if (lexicalLifetimeEnsured(asi) &&
```
Comment (nate-chandler) [inline lib/SILOptimizer/Transforms/SILMem2Reg.cpp:1181]: Should we add a predicate that combines these two conditions, considering how many times it's showing up?

File: lib/SILOptimizer/Transforms/SILMem2Reg.cpp [hunk 2]
```diff
@@ -105,9 +106,10 @@ class LiveValues {
       if (!lexicalLifetimeEnsured(asi)) {
         return stored;
       }
-      // We should have created a move of the @owned stored value.
-      assert(move);
-      return move;
+      assert(move || stored->getOwnershipKind() == OwnershipKind::None);
+      // If we had a trivial value stored in a lexical non trivial enum, we may
```
Comment (nate-chandler) [inline lib/SILOptimizer/Transforms/SILMem2Reg.cpp:110]: Should we have add a third kind `struct None` in addition to `struct Owned` and `struct Guaranteed`?

**Comment by meg-gupta** [comment]
Now included new None abstractions in Mem2Reg by @nate-chandler

**Comment by meg-gupta** [comment]
@swift-ci test

**Review by nate-chandler — APPROVED** [review]

**Comment by meg-gupta** [comment]
@swift-ci test

**Comment by meg-gupta** [comment]
@swift-ci test

**Pull request merged**